### PR TITLE
Adding yaml-cpp4rkt

### PR DESCRIPTION
**Pull request by allanleal**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/yaml-cpp4rkt/bld.bat [hunk 1]
```diff
@@ -0,0 +1,13 @@
+@REM Configure the build of phreeqc4rkt
```
Comment (tadeu) [inline recipes/yaml-cpp4rkt/bld.bat:1]:
```suggestion
@REM Configure the build of yaml-cpp4rkt
```

File: recipes/yaml-cpp4rkt/bld.bat [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+cmake -S . -B build ^
+    -DCMAKE_BUILD_TYPE=Release ^
+    -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
+    -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
+    -DBUILD_SHARED_LIBS=ON ^
+    -DYAML_BUILD_SHARED_LIBS=ON ^
+    -DYAML_CPP_BUILD_TESTS=OFF
+
+
+@REM Build and install phreeqc4rkt in %LIBRARY_PREFIX%
```
Comment (tadeu) [inline recipes/yaml-cpp4rkt/bld.bat:11]:
```suggestion
@REM Build and install yaml-cpp4rkt in %LIBRARY_PREFIX%
```

Comment (allanleal) [reply recipes/yaml-cpp4rkt/bld.bat:11]: This is the danger of copying and pasting recipes! :)

**Review by tadeu — APPROVED** [review]

File: recipes/yaml-cpp4rkt/build.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# due to many parallel jobs consuming all available memory
+JOBS=$((CPU_COUNT*2 - 1))
+
+echo "Using $JOBS parallel jobs out of $((CPU_COUNT*2)) available to build yaml-cpp4rkt."
+
+# Configure the build of yaml-cpp4rkt
+cmake -S . -B build ${CMAKE_ARGS} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_PREFIX_PATH=$PREFIX \
+    -DCMAKE_INSTALL_PREFIX=$PREFIX \
```
Comment (xhochy) [inline recipes/yaml-cpp4rkt/build.sh:15]: Part of `${CMAKE_ARGS}`
```suggestion
```

File: recipes/yaml-cpp4rkt/conda_build_config.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+MACOSX_SDK_VERSION:        # [osx]
```
Comment (xhochy) [inline recipes/yaml-cpp4rkt/conda_build_config.yaml:1]: Why is this needed? This isn't used in the `yaml-cpp` feedstock.

File: recipes/yaml-cpp4rkt/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+
+  run_constrained:
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx]
```
Comment (xhochy) [inline recipes/yaml-cpp4rkt/meta.yaml:23]: See above
```suggestion
```

File: recipes/yaml-cpp4rkt/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{% set version = "0.6.4.2.rkt" %}
```
Comment (xhochy) [inline recipes/yaml-cpp4rkt/meta.yaml:1]: I would rather keep the version numeric, you already have `rkt` in the name
```suggestion
{% set version = "0.6.4.2" %}
```

File: recipes/yaml-cpp4rkt/conda_build_config.yaml (hunk 4 above)
Comment (allanleal) [reply recipes/yaml-cpp4rkt/conda_build_config.yaml:1]: Inherited from other recipes in which this was needed. I'll remove this.

Comment (allanleal) [reply recipes/yaml-cpp4rkt/conda_build_config.yaml:1]: Removed!

File: recipes/yaml-cpp4rkt/conda_build_config.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+MACOSX_DEPLOYMENT_TARGET:  # [osx]
```
Comment (xhochy) [inline recipes/yaml-cpp4rkt/conda_build_config.yaml:1]: I meant that you probably don't need this file at all, just get rid of it ;)

Comment (allanleal) [reply recipes/yaml-cpp4rkt/conda_build_config.yaml:1]: Can I get rid of it once in the feedstock, please? ;)

Comment (xhochy) [reply recipes/yaml-cpp4rkt/conda_build_config.yaml:1]: No, this isn't needed so we shouldn't merge it at all.

Comment (dopplershift) [reply recipes/yaml-cpp4rkt/conda_build_config.yaml:1]: You'll just have another round of builds in the feedstock...

Comment (allanleal) [reply recipes/yaml-cpp4rkt/conda_build_config.yaml:1]: > No, this isn't needed so we shouldn't merge it at all.

Fair enough. I've removed it.
